test/lib/next-test-utils.ts

```diff
@@ -521,6 +521,17 @@ export async function killProcess(
       resolve()
     })
   })
+
+  await waitForCondition(2000, () => {
+    try {
+      process.kill(pid, 0)
+      // pid still exists, server still running
+      return false
+    } catch {
+      // kill throws an error if the pid doesn't exist
```

Comment:
wasn't sure if we needed these comments. I didn't think it was very clear why we're calling `process.kill` but maybe refactoring to `function checkIfPIDExists` would be better

Reply:
Also thinking about this a bit more, it might make more sense to move this check into `killApp` and then we could add a test that calls `killProcess` instead, checks that it hasn't finished until some task is complete, then checks to make sure it's been killed separately.

Reply:
refactored this and I think it's more clear now even without the comments